modules/plugin/chowdsp_plugin_utils/SharedUtils/chowdsp_GlobalPluginSettings.cpp

```diff
@@ -152,6 +158,7 @@ bool GlobalPluginSettings::loadSettingsFromFile()
         if (! JSONUtils::isSameType (value, newProperty))
         {
             // new property does not have the same type as the original!
+            // also hit when listenerFileChanged callback is hit and our properties are different then the settings file
```

Comment:
I'm not sure I fully understand... even if the locally stored properties are different from what's currently in the settings file, the types of each entry should still be the same?

Reply:
I was also confused about this, I should have left a more descriptive comment. 

From my understanding and what was happening when I was debugging this. When our local stored properties are misaligned with the file, some properties exist in the local settings, but not in the file, so when we hit this section of code, `value` is in `oldProperties` but when we try to get `name.data()` out of the newly updated file matching  `globalProperties`, it doesn't have a value for name.data() and returns null. This means the` value` is an int and `newProperty` is null, failing the if statement, int != null.

Thats what I have seen at least, let me know if that makes sense to you as well.

Thanks!

Reply:
Oh interesting... in that case I would have expected the line above `auto& newProperty = globalProperties[name.data()];` to either throw an exception or return an invalid object. But I guess it's nice that the `json` object "fails" in such a well-defined way.

Probably what we should do then is have an explicit error-handling path for that case. Like:
```cpp
auto& newProperty = globalProperties[name.data()];
if (newProperty.is_null())
{
    // The new properties loaded from the file do not have an entry for this field!
    jassertfalse;
    newProperty = value;
}
else if (! JSONUtils::isSameType (value, newProperty))
{
   // the same stuff here...
}
```

What do you think?

Reply:
Yup I think this is a good idea